src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  let temp = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'clear':
+        temp = {};
+        break;
+
+      case 'addProperties':
+        for (const addKey in actions[i].extraData) {
+          temp[addKey] = actions[i].extraData[addKey];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const rmKey of actions[i].keysToRemove) {
+          if (temp.hasOwnProperty(rmKey)) {
```

Comment:
Checking if an object has a property before deleting it is unnecessary, as the delete operation will fail silently if the property does not exist. You can safely delete the property without the check.